### PR TITLE
fix make requirements and update dependencies

### LINKED ISSUE #238: Bug: make requirements does not update Pipfile.lock when Pipfile has been manually
### What happened?

Running make requirements does not update the requirement files (requirements.txt and requirements-dev.txt) correctly if we have manually edited the Pipfile. This could happen if we are manually pinning a dependency or specifying some optional dependency.
Locking first could protect us from having outdated requirement files if we manually edit the Pipfile and would be a nice-to-have extra precaution.

### What steps did you take?

1. Update the Pipfile by adding a new package
2. Run make requirements to update the requirement files

### What behavior did you expect?

We expect the requirements files to include the latest changes made to the Pipfile.
This does not happen since we need to lock before running:

```
pipenv requirements > requirements.txt
pipenv requirements --dev > requirements-dev.txt
```
so we could instead have:

```
pipenv lock
pipenv requirements > requirements.txt
pipenv requirements --dev > requirements-dev.txt
```
This is because the Pipfile.lock is used to generate the requirement files

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by kairoaraujo**
- fix `make requirements`
- update the API requirements and Pipfile

Closes #238

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
       run: pip install tox tox-gh-actions
 
     - name: Run Python tests
-      run: tox
+      run: tox -r
```
Comment (KAUTH) [inline .github/workflows/ci.yml:27]: Nit:
```suggestion
      make tests
```

**Review by KAUTH — APPROVED** [review]
LGTM, just a small nit

File: .github/workflows/ci.yml (hunk 1 above)
Comment (MVrachev) [reply .github/workflows/ci.yml:27]: Does the GitHub workflow see `MAKEFILE`?

**Review by MVrachev — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kairoaraujo) [reply .github/workflows/ci.yml:27]: Yes, it does see it. 
But we need to refine and decide better about the Makefile structure. In our dev docs, we share the tox in our CI using make in some parts. It's the second time we think about it ([first one](https://github.com/vmware/repository-service-tuf-cli/pull/110#pullrequestreview-1159892929)). What is good.😄 
I will merge and we can discuss it more in the below issue:
- https://github.com/vmware/repository-service-tuf/issues/229

**Pull request merged**